js/components/operatorSignIn/operatorSelection.tsx

```diff
@@ -1,16 +1,40 @@
+import { ApiResult } from "../../api";
+import { findEmployeeByBadgeSerial } from "../../hooks/useEmployees";
+import { useNfc } from "../../hooks/useNfc";
+import { Employee } from "../../models/employee";
 import { className } from "../../util/dom";
-import { ReactElement, useId, useState } from "react";
+import { ReactElement, useEffect, useId, useState } from "react";
 
 export const OperatorSelection = ({
   nfcSupported: nfcSupported,
   onOK,
+  employees,
 }: {
   nfcSupported: boolean;
   onOK: (badge: string) => void;
+  employees: ApiResult<Employee[]>;
 }): ReactElement => {
   const inputId = useId();
   const [badgeEntry, setBadgeEntry] = useState<string>("");
 
+  const nfcAbortController = new AbortController();
```

Comment:
I forget, can these be reused?

Reply:
Excellent question, I'll have to look into it.

Reply:
Also I'm realizing that as written this will actually create a new AbortController on every render, which is not what I want. I'm working on a bit of a refactor of the `useNfc` hook now.

Reply:
OK! I will hang tight for that.

Reply:
Okay I refactored things a bit in 10c66ec and 4dcd61b to basically move the creation of the `AbortController` entirely into the `useNfc` hook to make sure it doesn't get reused. The logic is pretty tricky but I think I've got it working right, and given what a change this is I'm definitely going to do some manual testing.

Reply:
Okay, as best as I can tell this is working. Incidentally, [this](https://react.dev/learn/state-as-a-snapshot) and other parts of the React docs were helpful reminders about the component lifecycle.